src/Content/Conversation/Factory/Timeline.php

```diff
@@ -21,27 +21,48 @@
 
 namespace Friendica\Content\Conversation\Factory;
 
+use Friendica\Capabilities\ICanCreateFromTableRow;
 use Friendica\Content\Conversation\Collection\Timelines;
 use Friendica\Model\User;
 use Friendica\Content\Conversation\Entity\Timeline as TimelineEntity;
+use Friendica\Content\Conversation\Repository\Channel;
 use Friendica\Core\Config\Capability\IManageConfigValues;
 use Friendica\Core\L10n;
 use Friendica\Module\Conversation\Community;
 use Psr\Log\LoggerInterface;
 
-final class Timeline extends \Friendica\BaseFactory
+final class Timeline extends \Friendica\BaseFactory implements ICanCreateFromTableRow
 {
 	/** @var L10n */
 	protected $l10n;
 	/** @var IManageConfigValues The config */
 	protected $config;
+	/** @var Channel */
+	protected $channel;
 
-	public function __construct(L10n $l10n, LoggerInterface $logger, IManageConfigValues $config)
+	public function __construct(Channel $channel, L10n $l10n, LoggerInterface $logger, IManageConfigValues $config)
```

Comment:
This creates a dependency loop between `Factory/Timeline` and `Repository/Channel`, how does `DI` handle that, @nupplaphil ?

Reply:
Do you get any error running that code? I was mainly wondering how `DI` handles instancing these two classes since they depend on each other.

Reply:
No. It seems to work fine.